### ISSUE TITLE
MDNF ILLegal program detected

### ISSUE THREAD

**Issue by sheiy** [PlayCover/PlayCover#57]
[download link](https://dyp.li/GB8zkxURrFN)

<img width="1051" alt="image" src="https://user-images.githubusercontent.com/12210038/179439094-d130fc9a-58d6-4714-a968-fae3bf9c7076.png">

```
版本12.4（版号21F79）
Installing PlayTools
Copying PlayTools to Frameworks

System Integrity Protection status: enabled.

/Library/Developer/CommandLineTools

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "https://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>com.apple.security.app-sandbox</key>
    <true/>
    <key>com.apple.security.assets.movies.read-write</key>
    <true/>
    <key>com.apple.security.assets.music.read-write</key>
    <true/>
    <key>com.apple.security.assets.pictures.read-write</key>
    <true/>
    <key>com.apple.security.device.audio-input</key>
    <true/>
    <key>com.apple.security.device.bluetooth</key>
    <true/>
    <key>com.apple.security.device.camera</key>
    <true/>
    <key>com.apple.security.device.usb</key>
    <true/>
    <key>com.apple.security.files.downloads.read-write</key>
    <true/>
    <key>com.apple.security.files.user-selected.read-write</key>
    <true/>
    <key>com.apple.security.network.client</key>
    <true/>
    <key>com.apple.security.network.server</key>
    <true/>
    <key>com.apple.security.personal-information.addressbook</key>
    <true/>
    <key>com.apple.security.personal-information.calendars</key>
    <true/>
    <key>com.apple.security.personal-information.location</key>
    <true/>
    <key>com.apple.security.print</key>
    <true/>
    <key>com.apple.security.temporary-exception.sbpl</key>
    <array>
      <string>(allow user-preference-write (preference-domain ".GlobalPreferences"))</string>
      <string>(allow user-preference-read (preference-domain ".GlobalPreferences"))</string>
      <string>(allow file* file-read* file-write* file-write-data file-read-metadata file-ioctl (literal "/Users/sheiy/Library/Preferences/playcover.plist"))</string>
      <string>(allow file* file-read* file-read-metadata file-ioctl (subpath "/Users/sheiy/Library/Frameworks/865DN"))</string>
      <string>(allow file* file-read* file-read-metadata file-ioctl (subpath "/Users/sheiy/Library/Frameworks/E922N"))</string>
    </array>
  </dict>
</plist>

/Users/sheiy/Library/Containers/io.playcover.PlayCover/mhmobile.app: replacing existing signature

/usr/bin/vtool

/Users/sheiy/Library/Frameworks/PlayTools.framework/PlayTools:
Load command 50
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

/usr/bin/vtool

/Users/sheiy/Library/Containers/io.playcover.PlayCover/mhmobile.app/mhmobile:
Load command 66
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

Executable=/Users/sheiy/Library/Containers/io.playcover.PlayCover/mhmobile.app/mhmobile
Identifier=com.netease.mhxyhtb
Format=app bundle with Mach-O thin (arm64)
CodeDirectory v=20400 size=374892 flags=0x2(adhoc) hashes=11705+7 location=embedded
Signature=adhoc
Info.plist entries=52
TeamIdentifier=not set
Sealed Resources version=2 rules=10 files=2756
Internal requirements count=0 size=12

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "https://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>com.apple.security.app-sandbox</key>
    <true/>
    <key>com.apple.security.assets.movies.read-write</key>
    <true/>
    <key>com.apple.security.assets.music.read-write</key>
    <true/>
    <key>com.apple.security.assets.pictures.read-write</key>
    <true/>
    <key>com.apple.security.device.audio-input</key>
    <true/>
    <key>com.apple.security.device.bluetooth</key>
    <true/>
    <key>com.apple.security.device.camera</key>
    <true/>
    <key>com.apple.security.device.usb</key>
    <true/>
    <key>com.apple.security.files.downloads.read-write</key>
    <true/>
    <key>com.apple.security.files.user-selected.read-write</key>
    <true/>
    <key>com.apple.security.network.client</key>
    <true/>
    <key>com.apple.security.network.server</key>
    <true/>
    <key>com.apple.security.personal-information.addressbook</key>
    <true/>
    <key>com.apple.security.personal-information.calendars</key>
    <true/>
    <key>com.apple.security.personal-information.location</key>
    <true/>
    <key>com.apple.security.print</key>
    <true/>
    <key>com.apple.security.temporary-exception.sbpl</key>
    <array>
      <string>(allow user-preference-write (preference-domain ".GlobalPreferences"))</string>
      <string>(allow user-preference-read (preference-domain ".GlobalPreferences"))</string>
      <string>(allow file* file-read* file-write* file-write-data file-read-metadata file-ioctl (literal "/Users/sheiy/Library/Preferences/playcover.plist"))</string>
      <string>(allow file* file-read* file-read-metadata file-ioctl (subpath "/Users/sheiy/Library/Frameworks/PlayTools.framework"))</string>
    </array>
  </dict>
</plist>

/usr/bin/vtool

/Users/sheiy/Library/Frameworks/PlayTools.framework/PlayTools:
Load command 50
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

/usr/bin/vtool

/Users/sheiy/Library/Containers/io.playcover.PlayCover/mhmobile.app/mhmobile:
Load command 66
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

Executable=/Users/sheiy/Library/Containers/io.playcover.PlayCover/mhmobile.app/mhmobile
Identifier=com.netease.mhxyhtb
Format=app bundle with Mach-O thin (arm64)
CodeDirectory v=20400 size=374892 flags=0x2(adhoc) hashes=11705+7 location=embedded
Signature=adhoc
Info.plist entries=52
TeamIdentifier=not set
Sealed Resources version=2 rules=10 files=2756
Internal requirements count=0 size=12

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "https://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>com.apple.security.app-sandbox</key>
    <true/>
    <key>com.apple.security.assets.movies.read-write</key>
    <true/>
    <key>com.apple.security.assets.music.read-write</key>
    <true/>
    <key>com.apple.security.assets.pictures.read-write</key>
    <true/>
    <key>com.apple.security.device.audio-input</key>
    <true/>
    <key>com.apple.security.device.bluetooth</key>
    <true/>
    <key>com.apple.security.device.camera</key>
    <true/>
    <key>com.apple.security.device.usb</key>
    <true/>
    <key>com.apple.security.files.downloads.read-write</key>
    <true/>
    <key>com.apple.security.files.user-selected.read-write</key>
    <true/>
    <key>com.apple.security.network.client</key>
    <true/>
    <key>com.apple.security.network.server</key>
    <true/>
    <key>com.apple.security.personal-information.addressbook</key>
    <true/>
    <key>com.apple.security.personal-information.calendars</key>
    <true/>
    <key>com.apple.security.personal-information.location</key>
    <true/>
    <key>com.apple.security.print</key>
    <true/>
    <key>com.apple.security.temporary-exception.sbpl</key>
    <array>
      <string>(allow user-preference-write (preference-domain ".GlobalPreferences"))</string>
      <string>(allow user-preference-read (preference-domain ".GlobalPreferences"))</string>
      <string>(allow file* file-read* file-write* file-write-data file-read-metadata file-ioctl (literal "/Users/sheiy/Library/Preferences/playcover.plist"))</string>
      <string>(allow file* file-read* file-read-metadata file-ioctl (subpath "/Users/sheiy/Library/Frameworks/865DN"))</string>
      <string>(allow file* file-read* file-read-metadata file-ioctl (subpath "/Users/sheiy/Library/Frameworks/E922N"))</string>
    </array>
  </dict>
</plist>

/Users/sheiy/Library/Containers/io.playcover.PlayCover/mdnf.app: replacing existing signature

/usr/bin/vtool

/Users/sheiy/Library/Frameworks/PlayTools.framework/PlayTools:
Load command 50
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

/usr/bin/vtool

/Users/sheiy/Library/Containers/io.playcover.PlayCover/mdnf.app/mdnf:
Load command 73
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

Executable=/Users/sheiy/Library/Containers/io.playcover.PlayCover/mdnf.app/mdnf
Identifier=com.nexon.mdnf
Format=app bundle with Mach-O thin (arm64)
CodeDirectory v=20400 size=1194055 flags=0x2(adhoc) hashes=37304+7 location=embedded
Signature=adhoc
Info.plist entries=76
TeamIdentifier=not set
Sealed Resources version=2 rules=10 files=292
Internal requirements count=0 size=12

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "https://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>application-identifier</key>
    <string>NDX844788Z.com.nexon.mdnf</string>
    <key>aps-environment</key>
    <string>production</string>
    <key>com.apple.developer.applesignin</key>
    <array>
      <string>Default</string>
    </array>
    <key>com.apple.developer.associated-domains</key>
    <array>
      <string>applinks:k6dg.adj.st</string>
    </array>
    <key>com.apple.developer.team-identifier</key>
    <string>NDX844788Z</string>
  </dict>
</plist>

/Users/sheiy/Documents/PlayCoverInstall/tmp/79CA1C71-A132-4DFE-B503-D2B830B8DD55/Payload/mdnf.app: replacing existing signature

/usr/bin/otool

Replacing libswiftUIKit.dylib
/usr/bin/install_name_tool

/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: /Users/sheiy/Documents/PlayCoverInstall/tmp/79CA1C71-A132-4DFE-B503-D2B830B8DD55/Payload/mdnf.app/mdnf

Converting mdnf binary
/usr/bin/vtool

/Library/Developer/CommandLineTools/usr/bin/vtool warning: code signature will be invalid for /Users/sheiy/Documents/PlayCoverInstall/tmp/79CA1C71-A132-4DFE-B503-D2B830B8DD55/Payload/mdnf.app/mdnf

/Users/sheiy/Documents/PlayCoverInstall/tmp/79CA1C71-A132-4DFE-B503-D2B830B8DD55/Payload/mdnf.app/mdnf: replacing existing signature

/Users/sheiy/Library/Containers/io.playcover.PlayCover/mdnf.app: replacing existing signature

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "https://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>com.apple.security.app-sandbox</key>
    <true/>
    <key>com.apple.security.assets.movies.read-write</key>
    <true/>
    <key>com.apple.security.assets.music.read-write</key>
    <true/>
    <key>com.apple.security.assets.pictures.read-write</key>
    <true/>
    <key>com.apple.security.device.audio-input</key>
    <true/>
    <key>com.apple.security.device.bluetooth</key>
    <true/>
    <key>com.apple.security.device.camera</key>
    <true/>
    <key>com.apple.security.device.usb</key>
    <true/>
    <key>com.apple.security.files.downloads.read-write</key>
    <true/>
    <key>com.apple.security.files.user-selected.read-write</key>
    <true/>
    <key>com.apple.security.network.client</key>
    <true/>
    <key>com.apple.security.network.server</key>
    <true/>
    <key>com.apple.security.personal-information.addressbook</key>
    <true/>
    <key>com.apple.security.personal-information.calendars</key>
    <true/>
    <key>com.apple.security.personal-information.location</key>
    <true/>
    <key>com.apple.security.print</key>
    <true/>
    <key>com.apple.security.temporary-exception.sbpl</key>
    <array>
      <string>(allow user-preference-write (preference-domain ".GlobalPreferences"))</string>
      <string>(allow user-preference-read (preference-domain ".GlobalPreferences"))</string>
      <string>(allow file* file-read* file-write* file-write-data file-read-metadata file-ioctl (literal "/Users/sheiy/Library/Preferences/playcover.plist"))</string>
      <string>(allow file* file-read* file-read-metadata file-ioctl (subpath "/Users/sheiy/Library/Frameworks/PlayTools.framework"))</string>
    </array>
  </dict>
</plist>

/usr/bin/vtool

/Users/sheiy/Library/Frameworks/PlayTools.framework/PlayTools:
Load command 50
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

/usr/bin/vtool

/Users/sheiy/Library/Containers/io.playcover.PlayCover/mdnf.app/mdnf:
Load command 73
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

Executable=/Users/sheiy/Library/Containers/io.playcover.PlayCover/mdnf.app/mdnf
Identifier=com.nexon.mdnf
Format=app bundle with Mach-O thin (arm64)
CodeDirectory v=20400 size=1194855 flags=0x2(adhoc) hashes=37329+7 location=embedded
Signature=adhoc
Info.plist entries=76
TeamIdentifier=not set
Sealed Resources version=2 rules=10 files=293
Internal requirements count=0 size=12

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "https://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>com.apple.security.app-sandbox</key>
    <true/>
    <key>com.apple.security.assets.movies.read-write</key>
    <true/>
    <key>com.apple.security.assets.music.read-write</key>
    <true/>
    <key>com.apple.security.assets.pictures.read-write</key>
    <true/>
    <key>com.apple.security.device.audio-input</key>
    <true/>
    <key>com.apple.security.device.bluetooth</key>
    <true/>
    <key>com.apple.security.device.camera</key>
    <true/>
    <key>com.apple.security.device.usb</key>
    <true/>
    <key>com.apple.security.files.downloads.read-write</key>
    <true/>
    <key>com.apple.security.files.user-selected.read-write</key>
    <true/>
    <key>com.apple.security.network.client</key>
    <true/>
    <key>com.apple.security.network.server</key>
    <true/>
    <key>com.apple.security.personal-information.addressbook</key>
    <true/>
    <key>com.apple.security.personal-information.calendars</key>
    <true/>
    <key>com.apple.security.personal-information.location</key>
    <true/>
    <key>com.apple.security.print</key>
    <true/>
    <key>com.apple.security.temporary-exception.sbpl</key>
    <array>
      <string>(allow user-preference-write (preference-domain ".GlobalPreferences"))</string>
      <string>(allow user-preference-read (preference-domain ".GlobalPreferences"))</string>
      <string>(allow file* file-read* file-write* file-write-data file-read-metadata file-ioctl (literal "/Users/sheiy/Library/Preferences/playcover.plist"))</string>
      <string>(allow file* file-read* file-read-metadata file-ioctl (subpath "/Users/sheiy/Library/Frameworks/PlayTools.framework"))</string>
    </array>
  </dict>
</plist>

/usr/bin/vtool

/Users/sheiy/Library/Frameworks/PlayTools.framework/PlayTools:
Load command 50
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

/usr/bin/vtool

/Users/sheiy/Library/Containers/io.playcover.PlayCover/mdnf.app/mdnf:
Load command 73
      cmd LC_BUILD_VERSION
  cmdsize 24
 platform MACCATALYST
    minos 11.0
      sdk 14.0
   ntools 0

Executable=/Users/sheiy/Library/Containers/io.playcover.PlayCover/mdnf.app/mdnf
Identifier=com.nexon.mdnf
Format=app bundle with Mach-O thin (arm64)
CodeDirectory v=20400 size=1194855 flags=0x2(adhoc) hashes=37329+7 location=embedded
Signature=adhoc
Info.plist entries=76
TeamIdentifier=not set
Sealed Resources version=2 rules=10 files=293
Internal requirements count=0 size=12


```

**Comment by IsaacMarovitz** [comment]
This may be due to the game's anticheat

**Comment by ZhichGaming** [comment]
Have you tried enabling jailbreak bypass?

**Comment by sheiy** [comment]
@ZhichGaming if enable jailbreak bypass . the game will crash

**Comment by ohaiibuzzle** [comment]
https://github.com/PlayCover/PlayCover/issues/296#issuecomment-1237203209

TLDR we have not developed a bypass for NGSM yet, so, no, it just won't run.

**Comment by sheiy** [comment]
@ohaiibuzzle  Is it possible that it will work in the future?
